projects/question-editor-library/src/lib/components/question/question.component.ts

```diff
@@ -476,7 +503,47 @@ export class QuestionComponent implements OnInit, AfterViewInit {
     }
     this.toolbarConfig.title = questionTitle;
   }
+  output(event) {}
 
-}
+  onStatusChanges(event) {
+  }
 
+  valueChanges(event) {
+    this.childFormData = event;
+  }
+  validateFormFields() {
+    _.forEach(this.leafFormConfig, (formFieldCategory) => {
+      if (formFieldCategory.required && !this.childFormData[formFieldCategory.code]) {
+          this.showFormError = true;
+          this.toasterService.error('Please fill the required fields');
+          return false;
+      }
+    });
+    return true;
+  }
+  previewFormData(status) {
+    const formvalue = _.cloneDeep(this.leafFormConfig);
+    this.leafFormConfig = null;
+    _.forEach(formvalue, (formFieldCategory) => {
+      if (_.has(formFieldCategory, 'editable')) {
+      formFieldCategory.editable = status ? _.find(this.initialLeafFormConfig, {code: formFieldCategory.code}).editable : status;
+      formFieldCategory.default = this.childFormData[formFieldCategory.code];
+      }
+    });
+    this.leafFormConfig = formvalue;
+  }
+  populateFormData() {
+    this.childFormData = {};
+    _.forEach(this.leafFormConfig, (formFieldCategory) => {
+      if (this.questionMetaData && _.has(this.questionMetaData, formFieldCategory.code)) {
+        formFieldCategory.default = this.questionMetaData[formFieldCategory.code];
+        this.childFormData[formFieldCategory.code] = this.questionMetaData[formFieldCategory.code];
+      }
+    });
+  }
+  ngOnDestroy() {
+    this.onComponentDestroy$.next();
```

Comment:
looks like we are not using this `subject` anywhere. please use it.

Reply:
Done